spaceros/Earthfile

```diff
@@ -94,9 +102,18 @@ setup:
 
 sources:
   FROM +setup
-  COPY github.com/space-ros/space-ros:earthly-wrapper+repos-file/ros2.repos ros2.repos
+  RUN mkdir spaceros_repo_dir
+
+  IF[ -z "$(git ls-remote ${SPACEROS_REPO_URL} ${GIT_BRANCH})" ]
+    RUN echo "branch '${GIT_BRANCH}' does not exist in spaceros repo, cloning main branch"
+    GIT CLONE ${SPACEROS_REPO_URL} spaceros_repo_dir/
```

Comment:
These are good points. Let me break it down:

- *"Can we at least make a third case that detects if we're just using the main branch of the default repo and uses the githubusercontent CDN to grab the file?"*  - 
- For default spaceros repo, I would even go one step further and say that we should use CDN whenever `SPACEROS_REPO_URL="https://github.com/space-ros/space-ros.git"`, regardless of branch. I think that it is important, because can imagine that if we have multiple spaceros distros, we would rarely pull from main (you would rather pull from specific tag)
- *"I guess a more complete solution would be checking the url for common host platforms like GitHub and GitLab which expose files directly under a specific URL but that might be out of scope for this issue."* -> yes, but here we would ideally need to handle cases when such repos are private. Earthly's `GIT CLONE` handles that automagically (check [earthly docs](https://docs.earthly.dev/docs/guides/best-practices#git-clone-vs-run-git-clone))

In summary, I would refactor that so that we use CDN whenever we pull from default spaceros repo, and use GIT CLONE otherwise. I will document that in the code so that its clear for future generations :D.

Sounds good to you @Bckempa?

Reply:
No worries, I figured it out, I will post the PR today in the evening